### DISCUSSION TITLE
How to skip mfa setup completely from user login process ?

### DISCUSSION THREAD

**Discussion by 1swaraj** [Web3Auth/Web3Auth#317] [Help]
I don't run a very critical application so I would like my users to skip MFA

**Comment by himanshuchawla009** [comment]
Yes, you can skip mfa if you want. Web3Auth's OpenloginAdapter accepts `mfaLevel` parameter in constructor to configure the display of mfa screen:-

Currently following values are supported:-

- 'default' Setting mfa level to default will present mfa screen to user on every third login.
- 'optional' Setting mfa level to default will present mfa screen to user on every login but user can skip it.
- 'mandatory' Setting mfa level to mandatory will make it mandatory for user to setup mfa after login.
- 'none' Setting mfa level to none will make the user skip the mfa setup screen


You can read more about it here: https://web3auth.io/docs/sdk/web/core/usage#openloginloginparams

**Reply by aristidesgpg** [reply]
@himanshuchawla009 I set the mfaLevel as 'none', but it still shows the 2FA

**Reply by yashovardhan** [reply]
Additionally, please check: https://web3auth.io/docs/sdk/web/openlogin#multi-factor-authentication

Try to add mfaLevel in openlogin adapter

**Reply by aristidesgpg** [reply]
@yashovardhan Thanks.
It woks now

**Reply by aristidesgpg** [reply]
Can we hide 2fa in @web3auth/web3auth instead of @web3auth/core?

**Reply by yashovardhan** [reply]
@web3auth/web3auth is a child class of @web3auth/core, hence the same functions can work there as well.

The same way you have changed the mfaLevel in the openlogin adapter for. web3auth/core, you can use it for web3auth/web3auth.

**Reply by aristidesgpg** [reply]
But it doesn't work.
it says `walletWallet adpater is not ready yet`

**Reply by yashovardhan** [reply]
This error means that your initialisation function is not properly executed. Can you share your initModal function?

**Reply by aristidesgpg** [reply]
@yashovardhan I attached the code below

**Reply by aristidesgpg** [reply]
From the document, I can see that it doesn't use initModal to use openloginAdapter.

**Reply by yashovardhan** [reply]
The document only tells the code needed for openlogin adapter. Please don't consider it as a proper example. Refer to our [Integration Builder](https://web3auth.io/docs/integration-builder?lang=react&chain=eth&customAuthentication=yes&whitelabel=no&dynamicConstructorParams=no&usingEmailPasswordless=no&rnWorkflowMode=expo&evmFramework=web3) to see how it works

**Comment by shahbaz17** [comment]
For `web3auth/web3auth`, use the below code to disable the 2fa flow.

```js
const openloginAdapter = new OpenloginAdapter({
  loginSettings: {
    mfaLevel: "none",
  },
});
```

See https://web3auth.io/docs/sdk/web/openlogin#multi-factor-authentication

**Reply by shahbaz17** [reply]
@talentdev2020

**Reply by aristidesgpg** [reply]
```
@yashovardhan Now I don't use initModal because I can't find where I put the `mfaLevel` property.
So I used openlogin adapter

const openloginAdapter = new OpenloginAdapter({
		adapterSettings: {
			network: 'testnet',
			clientId,
			uxMode: 'popup',
			loginConfig: {
				google: {
					name: 'Custom Login',
					verifier: 'web3auth-verifier',
					typeOfLogin: 'jwt',
					clientId: process.env.NEXT_PUBLIC_GOOGLE_CLIENT_ID
				}
			}
		},
		loginSettings: {
			mfaLevel: 'none'
		}
	});

	web3auth.configureAdapter(openloginAdapter);
```

**Reply by aristidesgpg** [reply]
@shahbaz17 Yeah, I want to hide the 2FA in the web3auth/web3auth instead of web3auth core.  So I used openloginAdapter for web3auth/web3atuh, and used `connectTo()`, but it says walletWallet adapter is not ready yet error.
This wallet adapter not ready error is happening when using web3auth/core.  
Here is another discussion for it
https://github.com/Web3Auth/Web3Auth/discussions/400

**Reply by yashovardhan** [reply]
You need to use the initModal without any arguments. Using openlogin adapter doesn't mean you don't need to use initModal. Initialisation is an important step which is causing this error.

**Reply by aristidesgpg** [reply]
Ok, then should I use connect() or connectTo()?

**Reply by yashovardhan** [reply]
Use the following in this order
Instantiate Web3Auth
Instantiate Openlogin Adapter
web3auth.configureAdapter(openloginAdapter);
await web3auth.initModal();
await web3auth.connect();

**Reply by aristidesgpg** [reply]
Perfect!  I will try

**Reply by aristidesgpg** [reply]
@yashovardhan Awesome!
It works now!  Thanks

**Reply by aristidesgpg** [reply]
@yashovardhan After login, and logout, and try to login again, but this spinner modal doesn't appear.  When I didn't use openLoginAdapter, it worked.  I think this issue is related to #400 
<img width="432" alt="Screen Shot 2022-08-17 at 6 44 12 AM" src="https://user-images.githubusercontent.com/48382274/185149800-678149b1-e059-46c7-b1a2-46c67d106fbc.png">
